### ISSUE TITLE
FCM notifications not being received on iOS TestFlight specifically

### ISSUE THREAD

**Issue by cybex-dev** [ConnectyCube/connectycube-flutter-samples#154]
**TL;DR:** iOS app (code unchanged) uploaded to TestFlight doesn't receive FCM messages, thus callkit never activates and incoming call notifications aren't ever received. **_Occurs only only on iOS and only on TestFlight_**.

---

**Specify the sample to which the issue belongs (use [x]):**
[  ] Chat sample
[ x] P2P Call sample
[  ] Conference Call sample

**Platform (use [x])**
[ ] Android
[ x] iOS

**Device info**
Manufacture: iPhone
Model: 6s & 8
OS version: (13.6.1) & (14.4.1)

**Describe the bug:**
_Not a bug necessarily, not sure where the problem lies but this is the best place to start._

Firstly, Android runs perfectly, no problems to speak of, various builds local & on app store work perfectly!

**iOS is problematic**, specifically notifications.
- Local install (debug + release) [aps-environment: "production" or "development"] works perfectly.
- Upload to TestFlight: (release) [aps-environment: "production"], no notifications come through at all. ConnectyCube session notifications come through and show on screen by editing the code below:

**Code reference:** [call_manager.dart#L84](https://github.com/ConnectyCube/connectycube-flutter-samples/blob/master/p2p_call_sample/lib/src/managers/call_manager.dart#L84)

      } else if (callState == CallState.UNKNOWN) {
        // ConnectycubeFlutterCallKit.setCallState(sessionId: _currentCall.sessionId, callState: CallState.PENDING);
        // _showIncomingCallScreen(_currentCall);
        // if(Platform.isWindows || Platform.isMacOS){
        _showIncomingCallScreen(_currentCall!);                <-------- this line runs on Testflight, but the callkit doesn't register an incoming call
        // }
      }

**Steps to Reproduce:**
1. Clone repo
2. Create Firebase App
2. Create iOS BundleId with Push Notifications enabled & add bundleId to app
3. (background capabilities & push notification capability via ```aps-environment``` are already added)
4. Install GoogleServices.plist
5. Create APNS key on developer.apple site, added the downloaded .p8 file to firebase messaging
6. Create APS certificate (Sandbox & Prod SSL), extract .p12 and upload to ConnectyCube
7. (create icons - needed to upload to TestFlight)
8. (test &) Archive app in XCode, upload to TestFlight
9. Download on 2 iPhone (physical devices)
10. Sign in to accounts (assuming those exist on ConnectyCube)
11. Attempt to call `User X` from `User Y`, nothing happens. Do the reverse call, nothing happens. Close app, reopen, do the same procedure, nothing happens

(however, build locally in either debug|release mode with aps-environment development|production, calls **always** goes through on iOS & Android)

**Logs:**
```
No logs, can't get from TestFlight (failed multiple attempts in uploading a debug build, guidance would be appreciated if it could assist here)
```

**Actual result:**
Calling User X from User Y from TestFlight builds, no call notifications arrive on the called device (iOS specific)

**Expected behavior:**
Expecting FCM to notify the device of an incoming call via `backgroundMessageHandler()` or `onMessage()`, etc. which never happens, thus this function is never triggered (atleast on TestFlight, testing locally, this function is triggered when call notification arrives)

```swift
    // Handle incoming pushes
    public func pushRegistry(_ registry: PKPushRegistry,
                             didReceiveIncomingPushWith payload: PKPushPayload,
                             for type: PKPushType,
                             completion: @escaping () -> Swift.Void){
        
        FlutterVoipPushNotificationPlugin.didReceiveIncomingPush(with: payload, forType: type.rawValue)
        
        let signalType = payload.dictionaryPayload["signal_type"] as! String
        if(signalType == "endCall" || signalType == "rejectCall"){
            return
        }
        
        let uuid = payload.dictionaryPayload["session_id"] as! String
        let uID = payload.dictionaryPayload["caller_id"] as! Int
        let callerName = payload.dictionaryPayload["caller_name"] as! String
        let isVideo = payload.dictionaryPayload["call_type"] as! Int == 1;
        FlutterCallKitPlugin.reportNewIncomingCall(
            uuid,
            handle: String(uID),
            handleType: "generic",
            hasVideo: isVideo,
            localizedCallerName: callerName,
            fromPushKit: true
        )
        completion()
    }
```

**Additional info**

**_Been stuck on this for days, crunch time is approaching - I'm lost. Any info you need, testing, etc. I will perform immediately as this is hyper-crucial for me._** (sorry)

**Comment by cybex-dev** [comment]
**Preface:** 
the issue was mine.

**TL;DR**
changed only 1 reference of `CubeEnvironment` to `PRODUCTION`. 

---

There are multiple locations to change `CubeEnvironment`:

- [push_notifications_manager.dart#111](https://github.com/ConnectyCube/connectycube-flutter-samples/blob/master/p2p_call_sample/lib/src/managers/push_notifications_manager.dart#L111)
- [push_notifications_manager.dart#111](https://github.com/ConnectyCube/connectycube-flutter-samples/blob/master/p2p_call_sample/lib/src/managers/push_notifications_manager.dart#L253)
- [call_manager.dart#111](https://github.com/ConnectyCube/connectycube-flutter-samples/blob/master/p2p_call_sample/lib/src/managers/call_manager.dart#L186)

Suggestion to use, even better to add this in your "CallManagerService"'s `init()` method:
```dart
    bool isProduction = bool.fromEnvironment('dart.vm.product');
    parameters.environment = isProduction ? CubeEnvironment.PRODUCTION : CubeEnvironment.DEVELOPMENT;
```
---

**Debugging (process):**
The debugging process (being somewhat unfamiliar with Swift & XCode) could have been better. I considered various provisioning profiles, `aps-environment` settings, etc.

_**Since the issue only occurred on Testflight**, it made debugging alot more challenging and time consuming as uploading a debug build had its own set of issues_

Finally I added a bunch of logging, the one that was crucial was the [CB-SDK](https://pub.dev/documentation/connectycube_sdk/latest/connectycube_core/connectycube_core-library.html) debug entry (when a notification is received):

```json
[
  {
    "subscription": {
      "id": sub id,
      "_id": "insert sub id",
      "user_id": cube_user_id,
      "bundle_identifier": "insert bundle id",
      "client_identification_sequence": "insert client id",
      "notification_channel_id": 6,
      "udid": "insert-uuid",
      "platform_id": 1,
      "environment": "development",
      "notification_channel": {
        "name": "apns_voip"
      },
      "device": {
        "udid": "insert-uuid",
        "platform": {
          "name": "ios"
        }
      }
    }
  }
]
```
specifically, the following entry.
```json
environment": "development
```

This is due to APS used 2 different push notification environments, each with its own certificates (certificate is assigned to unique URL's where push notifications can come from). This, `aps-environment` is set to `'production` (see on upload Archive screen right before you start uploading) but I'm receiving `development` environment notifications - that needed fixing.

Reviewing my code, I finally found the issue (and fix mentioned above).
